File: MODUL 6/PRAK601-2310817210021-RAKHMAD FAUZHAN RAMADHAN.py
def buat_matriks(jumlah_baris, jumlah_kolom, elemen):
    matriks = []
    indeks = 0
    for i in range(jumlah_baris):
        baris = []
        for j in range(jumlah_kolom):
            baris.append(int(elemen[indeks]))
            indeks += 1
        matriks.append(baris)
    return matriks

def tampilkan_matriks(matriks):
    for baris in matriks:
        print(' '.join(map(str, baris)))

input_str1 = input()
input_str2 = input()  

jumlah_baris, jumlah_kolom = map(int, input_str1.split())
elemen = input_str2.split()

matriks = buat_matriks(jumlah_baris, jumlah_kolom, elemen)
tampilkan_matriks(matriks)

File: MODUL 6/PRAK601-2310817210021-RAKHMAD FAUZHAN RAMADHAN.c
#include <stdio.h>

void buatMatriks(int baris, int kolom, int matriks[baris][kolom], int elemen[]) {
    int index = 0;
    for (int i = 0; i < baris; i++) {
        for (int j = 0; j < kolom; j++) {
            matriks[i][j] = elemen[index++];
        }
    }
}

void tampilkanMatriks(int baris, int kolom, int matriks[baris][kolom]) {
    for (int i = 0; i < baris; i++) {
        for (int j = 0; j < kolom; j++) {
            printf("%d ", matriks[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int baris, kolom;
    scanf("%d %d", &baris, &kolom);

    int elemen[baris * kolom];
    for (int i = 0; i < baris * kolom; i++) {
        scanf("%d", &elemen[i]);
    }

    int matriks[baris][kolom];
    buatMatriks(baris, kolom, matriks, elemen);
    tampilkanMatriks(baris, kolom, matriks);

    return 0;
}
